Funcky.Test/Extensions/EnumerableExtensions/FailOnEnumerationList.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+
+namespace Funcky.Test.Extensions.EnumerableExtensions
+{
+    internal sealed class FailOnEnumerationList : IList<int>
+    {
+        private readonly int _length;
+
+        public FailOnEnumerationList(int length)
+            => _length = length;
+
+        public int Count
+            => _length;
+
+        public bool IsReadOnly
+            => true;
+
+        public int this[int index]
+        {
+            get => index >= 0 && index < _length ? index : throw new IndexOutOfRangeException();
+            set => throw new InvalidOperationException();
```

Comment:
Nitpick: I think `NotSupportedException` would be the correct thing to throw here

Reply:
Fixed in 55206ff34bd90b262b6c37e35a1cc64498b1935d